vizro-core/examples/scratch_dev/app.py

```diff
@@ -1,51 +1,254 @@
 
-from typing import Literal
+print("INITIALIZING")
 
-from dash import html
+SPECIES_COLORS = {"setosa": "#00b4ff", "versicolor": "#ff9222", "virginica": "#3949ab"}
+BAR_CHART_CONF = dict(x="species", color="species", color_discrete_map=SPECIES_COLORS)
+SCATTER_CHART_CONF = dict(x="sepal_length", y="petal_length", color="species", color_discrete_map=SPECIES_COLORS)
 
-import vizro.models as vm
-from vizro import Vizro
+# Relevant for the "page_6" only
+FILTER_COLUMN = "species"
+# FILTER_COLUMN = "sepal_length"
+# FILTER_COLUMN = "date_column"
 
 
-class Tooltip(vm.VizroBaseModel):
-    type: Literal["tooltip"] = "tooltip"
-    text: str = "This is a tooltip"
-
-    def build(self):
-        return html.Div(
-            [
-                dbc.Button("Button", id=f"{self.id}-tooltip-target"),
-                dbc.Tooltip(
-                    self.text,
-                    target=f"{self.id}-tooltip-target",
-                    is_open=True,
-                ),
-            ]
+def load_from_file(filter_column=FILTER_COLUMN, parametrized_species=None):
+    # Load the full iris dataset
+    df = px.data.iris()
+    df["date_column"] = pd.date_range(start=pd.to_datetime("2024-01-01"), periods=len(df), freq="D")
+
+    with open("data.yaml", "r") as file:
+        data = {
+            "setosa": 0,
+            "versicolor": 0,
+            "virginica": 0,
+            "min": 0,
+            "max": 10,
+            "date_min": "2024-01-01",
+            "date_max": "2024-05-29",
+        }
+        data.update(yaml.safe_load(file) or {})
+
+    if filter_column == "species":
+        df = pd.concat(
+            objs=[
+                df[df[filter_column] == "setosa"].head(data["setosa"]),
+                df[df[filter_column] == "versicolor"].head(data["versicolor"]),
+                df[df[filter_column] == "virginica"].head(data["virginica"]),
+            ],
+            ignore_index=True,
         )
+    elif filter_column == "sepal_length":
+        df = df[df[filter_column].between(data["min"], data["max"], inclusive="both")]
+    elif filter_column == "date_column":
+        date_min = pd.to_datetime(data["date_min"])
+        date_max = pd.to_datetime(data["date_max"])
+        df = df[df[filter_column].between(date_min, date_max, inclusive="both")]
+    else:
+        raise ValueError("Invalid FILTER_COLUMN")
+
+    if parametrized_species:
+        df = df[df["species"].isin(parametrized_species)]
+
+    return df
+
+
+data_manager["load_from_file"] = load_from_file
+data_manager["load_from_file_species"] = partial(load_from_file, filter_column="species")
+data_manager["load_from_file_sepal_length"] = partial(load_from_file, filter_column="sepal_length")
+data_manager["load_from_file_date_column"] = partial(load_from_file, filter_column="date_column")
+
+
+# TODO-DEV: Turn on/off caching to see how it affects the app.
+# data_manager.cache = Cache(config={"CACHE_TYPE": "SimpleCache", "CACHE_DEFAULT_TIMEOUT": 10})
```

Comment:
Just to check this is supposed to happen: Because we do not change an argument to the data loading function, the cache doesn't realise that the underlying data changes right?

Reply:
Correct. Underlying data is cached per data loading function argument which means that changing this argument (with the data_frame Parameter) creates a new cache item with the separate timeout.